### PR TITLE
Fix missing initializers in initializer lists

### LINKED ISSUE #50: Missing class initializers
Hi Marin,

There are a few classes that are missing class initializers. I will submit a pull request for this shortly.

Cheers

James

### PR DESCRIPTION
**Pull request by jamesattewell**
Hi Marin,

Fixes #50. Please review and merge the fixes. The main one that needed fixing was the evaluator one.

Cheers

James

### CODE REVIEW

**Review by m-peko — COMMENTED** [review]
Actually, these initializers are not necessary because default constructors of complex data members are called implicitly.

**Comment by jamesattewell** [comment]
Hi Marin,

I have reverted the complex type initializers as they are not really needed.

Cheers

James

**Comment by m-peko** [comment]
Can you please make this to be only 1 commit instead of 2? So that we have clean git history.

File: src/evaluator.cpp [hunk 1]
```diff
@@ -32,16 +32,12 @@
...
+    : is_activated_(false)
 {}
 
 evaluator::evaluator(field_map const& fields)
-	: is_activated_(false),
-	  result_visitor_(),
-	  expression_tree_() {
-    result_visitor_.fields(fields);
+	: is_activated_(false) {
+		result_visitor_.fields(fields);
```
Comment (m-peko) [inline src/evaluator.cpp:40]: Please, pay attention to indentation. It should be 4 spaces.

Comment (jamesattewell) [reply src/evaluator.cpp]: Sorry Marin, Visual Studio had defaulted back to using tabs for some reason. All sorted now.

**Pull request merged**